packages/iov-stream/src/valueandupdates.ts

```diff
@@ -19,11 +23,19 @@ export class ValueAndUpdates<T> {
     this.updates = MemoryStream.createWithMemory(this.producer);
   }
 
-  public waitFor(value: T): Promise<void> {
+  /**
+   * Resolves as soon as search value is found.
+   *
+   * @param search either a value or a function that must return true when found
+   */
+  public waitFor(search: SearchFunction<T> | T): Promise<void> {
+    const searchImplementation: SearchFunction<T> =
+      typeof search === "function" ? (search as SearchFunction<T>) : (value: T): boolean => value === search;
+
     return new Promise((resolve, reject) => {
       const subscription = this.updates.subscribe({
         next: newValue => {
-          if (newValue === value) {
+          if (searchImplementation(newValue)) {
```

Comment:
Since you seem to be updating this code a bit, how about handling error case here as well with a reject?

Reply:
Which error do you mean?

Reply:
The stream subscription can call
Next (n times) 
Completed (0 or 1)
Error (0 or 1)

If error is called, we don't reject the promise